### PR TITLE
Add back support to obtain more wifi stats

### PR DESCRIPTION
**Pull request by jwei5**
- Add back changes from PR6439 and modify to use new structure trwifi_signal_quality
- Add new APIs to obtain more wifi stats from driver, such as tx/rx stats, deauth reason, km4 lib version
- Modify and add structures in wifi.h to support obtaining more wifi stats

### CODE REVIEW

File: os/include/tinyara/net/if/wifi.h [hunk 1]
```diff
@@ -245,8 +250,28 @@ typedef struct {
 	trwifi_status_e wifi_status;    /**<  @ref trwifi_status                     */
 	struct timeval disconn_time;    // optional: store the last time when the connection is disconnected
 	int32_t reason_code;            // optional: the reason why connection is disconnected. value is vendor specific
+	char bssid[TRWIFI_MACADDR_LEN]; /**< bssid is a mac address of AP */
```
Comment (hk-gwak) [inline os/include/tinyara/net/if/wifi.h:253]: Hi @jwei5, Could you remove this line?
```suggestion

```

Comment (jwei5) [reply os/include/tinyara/net/if/wifi.h]: Ok, have updated it

**Comment by hk-gwak** [comment]
In https://github.com/Samsung/TizenRT/pull/6349, there is a possibility of compatibility issues due to API changes. So changed to only add API in this PR.

**Review by hk-gwak — APPROVED** [review]
LGTM

**Comment by hk-gwak** [comment]
Easysetup test completed on both AI dual and AI Lite board.

**Pull request merged**